tests/test_workers/test_celery_logging.py

```diff
@@ -34,39 +77,76 @@ def _dummy_task(msg, request_id):
     try:
         celery_logging.setup_task_logging(task_id, task, args=["hello"], kwargs={"request_id": 3})
         workers_logger.info("Test log message")
-        assert "#%(request_id)s" in color_handler.formatter._fmt
     finally:
-        root_logger.removeHandler(color_handler)
+        for handler in logger.handlers:
+            if isinstance(handler, logging.FileHandler):
+                logger.removeHandler(handler)
+
+    # verify that correct message was logged
+    with open(os.path.join(str(tmpdir), "3.log")) as f:
```

Comment:
Optional: This might work
```suggestion
    with open(tmpdir.join("3.log")) as f:
```